contracts/0.8.9/StakingRouter.sol

```diff
@@ -0,0 +1,922 @@
+        uint16 treasuryFee;
+        /// @notice target percent of total keys in protocol, in BP
+        uint16 targetShare;
+        /// @notice staking module status if staking module can not accept the deposits or can participate in further reward distribution
+        uint8 status;
+        /// @notice name of staking module
+        string name;
+        /// @notice block.timestamp of the last deposit of the staking module
+        uint64 lastDepositAt;
+        /// @notice block.number of the last deposit of the staking module
+        uint256 lastDepositBlock;
+        /// @notice number of exited keys
+        uint256 exitedKeysCount;
+    }
+
+    struct StakingModuleCache {
+        address stakingModuleAddress;
+        uint24 stakingModuleId;
+        uint16 stakingModuleFee;
+        uint16 treasuryFee;
+        uint16 targetShare;
+        StakingModuleStatus status;
+        uint256 activeKeysCount;
+        uint256 availableKeysCount;
+    }
+
+    bytes32 public constant MANAGE_WITHDRAWAL_CREDENTIALS_ROLE = keccak256("MANAGE_WITHDRAWAL_CREDENTIALS_ROLE");
+    bytes32 public constant STAKING_MODULE_PAUSE_ROLE = keccak256("STAKING_MODULE_PAUSE_ROLE");
+    bytes32 public constant STAKING_MODULE_RESUME_ROLE = keccak256("STAKING_MODULE_RESUME_ROLE");
+    bytes32 public constant STAKING_MODULE_MANAGE_ROLE = keccak256("STAKING_MODULE_MANAGE_ROLE");
+    bytes32 public constant REPORT_EXITED_KEYS_ROLE = keccak256("REPORT_EXITED_KEYS_ROLE");
+    bytes32 public constant UNSAFE_SET_EXITED_KEYS_ROLE = keccak256("UNSAFE_SET_EXITED_KEYS_ROLE");
+    bytes32 public constant REPORT_REWARDS_MINTED_ROLE = keccak256("REPORT_REWARDS_MINTED_ROLE");
+
+    bytes32 internal constant LIDO_POSITION = keccak256("lido.StakingRouter.lido");
+
+    /// @dev Credentials which allows the DAO to withdraw Ether on the 2.0 side
+    bytes32 internal constant WITHDRAWAL_CREDENTIALS_POSITION = keccak256("lido.StakingRouter.withdrawalCredentials");
+
+    /// @dev total count of staking modules
+    bytes32 internal constant STAKING_MODULES_COUNT_POSITION = keccak256("lido.StakingRouter.stakingModulesCount");
+    /// @dev id of the last added staking module. This counter grow on staking modules adding
+    bytes32 internal constant LAST_STAKING_MODULE_ID_POSITION = keccak256("lido.StakingRouter.lastStakingModuleId");
+    /// @dev mapping is used instead of array to allow to extend the StakingModule
+    bytes32 internal constant STAKING_MODULES_MAPPING_POSITION = keccak256("lido.StakingRouter.stakingModules");
+    /// @dev Position of the staking modules in the `_stakingModules` map, plus 1 because
+    ///      index 0 means a value is not in the set.
+    bytes32 internal constant STAKING_MODULE_INDICES_MAPPING_POSITION = keccak256("lido.StakingRouter.stakingModuleIndicesOneBased");
+
+    uint256 public constant FEE_PRECISION_POINTS = 10 ** 20; // 100 * 10 ** 18
+    uint256 public constant TOTAL_BASIS_POINTS = 10000;
```

Comment:
We don't need to precisions here. No reason to set the fees in constructor with 1e4 precision, and operate further with 1e20. Especially if remove obsolete TOTAL_BASIS_POINTS from Lido contract.

Reply:
I guess we decided to leave it as is